client/ayon_shotgrid/tray/credential_dialog.py

```diff
@@ -0,0 +1,221 @@
+        self.error_label.setStyleSheet("color: red;")
+        self.error_label.setWordWrap(True)
+        self.error_label.hide()
+
+        self.input_layout = QtWidgets.QFormLayout()
+        self.input_layout.setContentsMargins(10, 15, 10, 5)
+
+        self.input_layout.addRow(self.url_label, self.url_input)
+        self.input_layout.addRow(self.login_label, self.login_input)
+        self.input_layout.addRow(self.password_label, self.password_input)
+        self.input_layout.addRow(self.error_label)
+
+        self.login_button = QtWidgets.QPushButton("Login")
+        self.login_button.setToolTip("Log in shotgrid instance")
+        self.login_button.clicked.connect(self._on_shotgrid_login_clicked)
+
+        self.logout_button = QtWidgets.QPushButton("Logout")
+        self.logout_button.setToolTip("Log out shotgrid instance")
+        self.logout_button.clicked.connect(self._on_shotgrid_logout_clicked)
+
+        if self._is_logged:
+            self.login_button.setDisabled(True)
+        else:
+            self.logout_button.setDisabled(True)
+
+        self.buttons_layout = QtWidgets.QHBoxLayout()
+        self.buttons_layout.addWidget(self.logout_button)
+        self.buttons_layout.addWidget(self.login_button)
+
+        self.main_widget = QtWidgets.QVBoxLayout(self)
+        self.main_widget.addLayout(self.input_layout)
+        self.main_widget.addLayout(self.buttons_layout)
+        self.setLayout(self.main_widget)
+
+    def show(self, *args, **kwargs):
+        super(CredentialsDialog, self).show(*args, **kwargs)
+        self._fill_shotgrid_url()
+        self._fill_shotgrid_login()
+
+    def _fill_shotgrid_url(self):
+        servers = {
+            "server_one": {
+                "shotgrid_url": self._module.module_settings.get(
+                    'shotgrid_server',
+                    ''
+                )
+            }
+        }
+
+        if servers:
+            for _, v in servers.items():
+                self.url_input.addItem("{}".format(v.get('shotgrid_url')))
+            self._valid_input(self.url_input)
+            self.login_button.show()
+            self.logout_button.show()
+            enabled = True
+        else:
+            self.set_error("Ask your admin to add shotgrid server in settings")
+            self._invalid_input(self.url_input)
+            self.login_button.hide()
+            self.logout_button.hide()
+            enabled = False
+
+        self.login_input.setEnabled(enabled)
+        self.password_input.setEnabled(enabled)
+
+    def _fill_shotgrid_login(self):
+        login = credentials.get_local_login()
+
+        if login:
+            self.login_input.setText(login)
+
+    def _clear_shotgrid_login(self):
+        self.login_input.setText("")
+        self.password_input.setText("")
+
+    def _on_shotgrid_login_clicked(self):
+        login = self.login_input.text().strip()
+        password = self.password_input.text().strip()
```

Comment:
I would say that white characters are valid entries for password.
```suggestion
        password = self.password_input.text()
```